src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,150 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.Ui;
+import duke.commands.AddCommand;
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.FindCommand;
+import duke.commands.HelpCommand;
+import duke.commands.ListCommand;
+import duke.exception.InvalidCommand;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.ToDo;
+
+
+/**
+ * Represents parser to parse and create appropriate Command object to handle
+ * the commands given by user.
+ *
+ */
+public class Parser {
+    public Parser() {
+    }
+
+    /**
+     * Generates the correct Command object to be executed.
+     *
+     * @param command User's input.
+     * @return Appropriate command to execute logic of input.
+     */
+    public static Command parse(String command) {
+        try {
+            if (command.contains("done")) {
+                String[] doneInputs = command.split(" ");
```

Comment:
I notice this method code is very long. Will it be look nicer if you handle each condition by using another method instead of putting all of them in one method? 🤔

Reply:
I agree that this is a good suggestion, will work on it to improve it! 😃